app/controllers/api/flavors_controller.rb

```diff
@@ -1,4 +1,28 @@
 module Api
   class FlavorsController < BaseController
+    def create_resource(_type, _id, data)
+      manager_id = parse_id(data['ems'], :providers)
```

Comment:
Since you require an `ems` attribute, I'm wondering if it would be more natural to create flavors as a subcollection of a provider, i.e.:

```
POST /api/providers/:pid/flavors
```

That should simplify things - you don't need special handling in the event that you don't have an `ems` attribute, and it helps keep things consistent (we call ems's "providers" in the API). Each time we add a new required attribute with a new name, it needs some documentation on how to use it. Whereas the workflow for creating subcollection resources is already well documented. 

/cc @abellotti any thoughts?

Reply:
If flavors are only created within the scope of a provider, then yes, that makes sense @imtayadeway and more elegant.

Reply:
@imtayadeway Hi, thank you for advice and review! Where can I find documentation for creating subcollection resources?

Reply:
@alexander-demichev there a lots of examples - here's one I picked at random: https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/subcollections/snapshots.rb#L8-L23

This way you can create a snapshot of a vm/instance. It happens to do this asynchronously, but you probably don't need to do that. The most important part is the naming - once you've added flavors as a subcollection of providers in the api.yml - there's some metaprograming that routes the message to a `flavors_create_resource` method that you'll need to create with the same signature as above. That should be enough to get started - ping me on gitter if you need any other help. Thanks!